tests/test_platform_config.py

```diff
@@ -0,0 +1,211 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for PlatformConfig module."""
+
+import os
+import sys
+
+import pytest
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from triton_kernel_agent.platform_config import (
+    DEFAULT_PLATFORM,
+    PLATFORMS,
+    get_platform,
+    get_platform_choices,
+)
+
+
+class TestGetPlatform:
+    """Tests for the get_platform() function."""
+
+    def test_get_platform_cuda(self):
+        """get_platform('cuda') should return CUDA config."""
+        config = get_platform("cuda")
+        assert config.name == "cuda"
+        assert config.device_string == "cuda"
+
+    def test_get_platform_xpu(self):
+        """get_platform('xpu') should return XPU config."""
+        config = get_platform("xpu")
+        assert config.name == "xpu"
+        assert config.device_string == "xpu"
+
+    def test_get_platform_invalid_raises_valueerror(self):
+        """get_platform() with invalid name should raise ValueError."""
+        with pytest.raises(ValueError) as exc_info:
+            get_platform("invalid_platform")
+        assert "Unknown platform" in str(exc_info.value)
+        assert "invalid_platform" in str(exc_info.value)
+
+    def test_get_platform_error_shows_available(self):
+        """ValueError message should list available platforms."""
+        with pytest.raises(ValueError) as exc_info:
+            get_platform("nonexistent")
+        error_msg = str(exc_info.value)
+        assert "cuda" in error_msg
+        assert "xpu" in error_msg
+
+    def test_get_platform_case_sensitive(self):
+        """Platform names should be case-sensitive."""
+        with pytest.raises(ValueError):
+            get_platform("CUDA")
+        with pytest.raises(ValueError):
+            get_platform("Xpu")
+
+
+class TestGetPlatformChoices:
+    """Tests for the get_platform_choices() function."""
+
+    def test_get_platform_choices_contains_cuda_and_xpu(self):
+        """Choices should include both cuda and xpu."""
+        choices = get_platform_choices()
+        assert "cuda" in choices
+        assert "xpu" in choices
+
+    def test_get_platform_choices_is_sorted(self):
+        """Choices should be sorted alphabetically."""
+        choices = get_platform_choices()
+        assert choices == sorted(choices)
+
+    def test_get_platform_choices_matches_platforms_registry(self):
+        """Choices should match PLATFORMS registry keys."""
+        choices = get_platform_choices()
+        assert set(choices) == set(PLATFORMS.keys())
+
+
+class TestPlatformConfigs:
+    """Tests for CUDA and XPU platform configurations."""
+
+    def test_cuda_device_string(self):
+        """CUDA device_string should be 'cuda'."""
+        config = get_platform("cuda")
+        assert config.device_string == "cuda"
```

Comment:
subset of  `TestGetPlatform.test_get_platform_cuda`

Reply:
Done